util/compress/deflate_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package compress
+
+import (
+	"bytes"
+	"compress/gzip"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestTrivialCompression(t *testing.T) {
+	bufLen := 10240
+	buffer := make([]byte, bufLen)
+	for i := range buffer {
+		buffer[i] = byte(i % 256)
+	}
+
+	compressedBuffer := make([]byte, 0, bufLen)
+	len, compressedOutput, err := Compress(buffer, compressedBuffer, 9)
+	require.NoError(t, err)
+	require.NotZero(t, len)
+	require.Equal(t, compressedBuffer[:len], compressedOutput)
+
+	decompressedBuffer := make([]byte, 0, bufLen)
+	decompressedOutput, err := Decompress(compressedOutput, decompressedBuffer)
+	require.NoError(t, err)
+	require.Equal(t, decompressedOutput, buffer)
+}
+
+func BenchmarkCompression(b *testing.B) {
+	bufLen := 1024000
```

Comment:
do we know how the compression algorithms compare with bufLen around 10000 (around the size of 100 txns)

Reply:
compression algorithms are heuristic based, so in order to get a better sense, we have to test it with some "real" data.